src/plugins/intel_gpu/tests/test_cases/cl_mem_input_test.cpp

```diff
@@ -105,9 +106,9 @@ TEST(cl_mem_check, check_2_inputs) {
 
     device_query query(engine_types::ocl, runtime_types::ocl, static_cast<void*>(ocl_instance->_context.get()));
     auto devices = query.get_available_devices();
-
-    auto engine_config = cldnn::engine_configuration();
-    auto engine = engine::create(engine_types::ocl, runtime_types::ocl, devices.begin()->second, engine_config);
+    auto iter = devices.find(debug_configuration::get_device_id());
+    auto& device = iter != devices.end() ? iter->second : devices.begin()->second;
```

Comment:
Looks like such kind of tests may require some additional work..
1. test binary should be capable of reporting detected device IDs.
2. eventually, we'll allow detection of gpus from other vendors (see #12577), so this approach with devices detection in external context won't work well if we provide all visible devices from multiple contexts to user.

So maybe this interop logic should be inverted, i.e. we create engine for specified device id and then extract native ocl handles and assign them to `ocl_instance`

Reply:
Done.